src/PyObjectProxyHandler.cc

```diff
@@ -0,0 +1,123 @@
+
+  PyObject *nonDunderKeys = PyList_New(0);
+  for (size_t i = 0; i < keysLength; i++) {
+    PyObject *key = PyList_GetItem(keys, i);
+    if (Py_IsFalse(PyObject_CallMethod(key, "startswith", "(s)", "__"))) { // if key starts with "__", ignore it
+      PyList_Append(nonDunderKeys, key);
+    }
+  }
+
+  size_t length = PyList_Size(nonDunderKeys);
+
+  return handleOwnPropertyKeys(cx, nonDunderKeys, length, props);
+}
+
+bool PyObjectProxyHandler::delete_(JSContext *cx, JS::HandleObject proxy, JS::HandleId id,
+  JS::ObjectOpResult &result) const {
+  PyObject *attrName = idToKey(cx, id);
+  PyObject *self = JS::GetMaybePtrFromReservedSlot<PyObject>(proxy, PyObjectSlot);
+  if (PyObject_SetAttr(self, attrName, NULL) < 0) {
+    return result.failCantDelete(); // raises JS exception
+  }
+  return result.succeed();
+}
+
+bool PyObjectProxyHandler::has(JSContext *cx, JS::HandleObject proxy, JS::HandleId id,
+  bool *bp) const {
+  return hasOwn(cx, proxy, id, bp);
+}
+
+bool PyObjectProxyHandler::getOwnPropertyDescriptor(
+  JSContext *cx, JS::HandleObject proxy, JS::HandleId id,
+  JS::MutableHandle<mozilla::Maybe<JS::PropertyDescriptor>> desc
+) const {
+  PyObject *attrName = idToKey(cx, id);
+  PyObject *self = JS::GetMaybePtrFromReservedSlot<PyObject>(proxy, PyObjectSlot);
+  PyObject *item = PyObject_GetAttr(self, attrName);
+  if (!item) { // clear error, we will be returning undefined in this case
+    PyErr_Clear();
+  }
+
+  return handleGetOwnPropertyDescriptor(cx, id, desc, item);
+}
+
+bool PyObjectProxyHandler::set(JSContext *cx, JS::HandleObject proxy, JS::HandleId id,
+  JS::HandleValue v, JS::HandleValue receiver,
+  JS::ObjectOpResult &result) const {
+  JS::RootedValue *rootedV = new JS::RootedValue(cx, v);
+  PyObject *attrName = idToKey(cx, id);
+  JS::RootedObject *global = new JS::RootedObject(cx, JS::GetNonCCWObjectGlobal(proxy));
+  PyObject *self = JS::GetMaybePtrFromReservedSlot<PyObject>(proxy, PyObjectSlot);
+  if (PyObject_SetAttr(self, attrName, pyTypeFactory(cx, global, rootedV)->getPyObject())) {
+    return result.failCantSetInterposed(); // raises JS exception
+  }
+  return result.succeed();
+}
+
+bool PyObjectProxyHandler::enumerate(JSContext *cx, JS::HandleObject proxy,
+  JS::MutableHandleIdVector props) const {
+  return this->ownPropertyKeys(cx, proxy, props);
+}
+
+bool PyObjectProxyHandler::hasOwn(JSContext *cx, JS::HandleObject proxy, JS::HandleId id,
+  bool *bp) const {
+  PyObject *attrName = idToKey(cx, id);
+  PyObject *self = JS::GetMaybePtrFromReservedSlot<PyObject>(proxy, PyObjectSlot);
+  *bp = PyObject_HasAttr(self, attrName) == 1;
+  return true;
+}
+
+bool PyObjectProxyHandler::getOwnEnumerablePropertyKeys(
+  JSContext *cx, JS::HandleObject proxy,
+  JS::MutableHandleIdVector props) const {
+  return this->ownPropertyKeys(cx, proxy, props);
+}
+
+void PyObjectProxyHandler::finalize(JS::GCContext *gcx, JSObject *proxy) const {
+  return handleFinalize(proxy);
```

Comment:
just don't override the method at all and let it inherit the parent class version

Reply:
done